skrifa/src/outline/glyf/hint/engine/outline.rs

```diff
@@ -0,0 +1,218 @@
+//! Managing outlines.
+//!
+//! Implements 4 instructions.
+//!
+//! See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#managing-outlines>
+
+use super::{
+    super::graphics_state::{CoordAxis, ZonePointer},
+    Engine, OpResult,
+};
+
+impl<'a> Engine<'a> {
+    /// Flip point.
+    ///
+    /// FLIPPT[] (0x80)
+    ///
+    /// Pops: p: point number (uint32)
+    ///
+    /// Uses the loop counter.
+    ///
+    /// Flips points that are off the curve so that they are on the curve and
+    /// points that are on the curve so that they are off the curve. The point
+    /// is not marked as touched. The result of a FLIPPT instruction is that
+    /// the contour describing part of a glyph outline is redefined.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#flip-point>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L5002>
+    pub(super) fn op_flippt(&mut self) -> OpResult {
+        let count = self.graphics_state.loop_counter as usize;
+        self.graphics_state.loop_counter = 1;
+        // In backward compatibility mode, don't flip points after IUP has
+        // been done.
+        if self.graphics_state.backward_compatibility
```

Comment:
In this context it's not entirely obvious what backward_compatibility would be with, I wonder if we should name it to more clearly indicate? For example, `if self.graphics_state.freetype_compat` makes it very clear what we're compatible with.

Reply:
This isn’t a FT compatibility thing. It’s a feature of the MS rasterizer that allows fonts (or specific glyphs) to opt in to cleartype native mode. Basically, both MS and FT implement a bunch of hacks for non cleartype fonts that prevent bytecode from modifying outlines in certain ways. This flag (which takes its name from FT) controls that behavior and is toggled by the INSTCTRL instruction.

Reply:
Thank you, that's helpful. I suggest putting some of that on the comment for the backward compatibility flag.